utils/camino_sorting.go

```diff
@@ -0,0 +1,23 @@
+// Copyright (C) 2023, Chain4Travel AG. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package utils
+
+import "math"
+
+// Returns true if the elements in [s] are unique and sorted.
+// math.MaxUint32 is skipped because they are used as wildcard
+func IsSortedAndUniqueOrderedSigIndices(s []uint32) bool {
+	var last uint32
+	for i := 0; i < len(s)-1; i++ {
+		if s[i] != math.MaxUint32 {
+			last = s[i]
+		}
+		if s[i+1] != math.MaxUint32 {
+			if last >= s[i+1] {

```

Comment:
`>=` Is it allows duplicates?

I can see it this way (maybe simpler)
This particular check is valid whether (one of the following applies):
- all indices are MaxUint32
- satisfies "old check" `utils.IsSortedAndUniqueOrdered(in.SigIndices)`

Reply:
This function "skips" math.Uint32, but still makes sure that "non-math.Uint32) are a.) unique and b.) sorted

0,1,2,M,2,3 wold fail because `last` is 2 and s[i+1] is also 2

Reply:
🤦 My fault the duplicate argument is wrong.

I mean if the input is MSig alias all indices will be `MaxInt` otherwise we can do the "old test".